lib/std/zig/system.zig

```diff
@@ -89,7 +89,13 @@ pub const NativePaths = struct {
         }
 
         if (!is_windows) {
-            const triple = try Target.current.linuxTriple(allocator);
+            // on linux the native abi might be different if we are statically linked
+            const target = if (Target.current.os.tag == .linux)
+                (try NativeTargetInfo.detectAbiAndDynamicLinker(allocator, Target.current.cpu, Target.current.os, .{})).target
+            else
+                Target.current;
+
+            const triple = try target.linuxTriple(allocator);
```

Comment:
`linuxTriple` is a pretty bad name for a function that has nothing to do with Linux heh
This code path is executed for every non-Windows target...

Also, the uses of `Target.current` below can be replaced with the newly introduced parameter, for consistency sake.